cmd/catchupsrv/download.go

```diff
@@ -44,34 +48,83 @@ var connsFlag = flag.Int("conns", 2, "Number of connections per server")
 var serverList []string
 var nextBlk uint64
 
-func blockToString(blk uint64) string {
-	return strconv.FormatUint(blk, 36)
+// padLeftZeros pad the string s with zeros on the left to the length n
+func padLeftZeros(s string, n int) string {
+	return strings.Repeat("0", n-len(s)) + s
```

Comment:
`strings.Repeat` panics if the second argument is negative.  The second argument can be negative as a result of an HTTP request that supplies a round number greater than 2176782336.  It would be good not to panic on bad HTTP inputs.

Reply:
Thanks! Fixed. There is a case the code does not handle very cleanly though: when the block number is above 36**6 (which should only happens in more than 350 years), the block `abcdefg` would be in `ab/cd/abcdefg` which is a bit confusing (it should be in `abc/de/abcdefg`.

Reply:
What about changing `blockToPath` so that instead of using `s[0:2]` and `s[2:4]`, it uses `s[len(s)-6:len(s)-4]` and `s[len(s)-4:len(s)-2]` respectively?  That'll make the directory hashing slightly more natural in this corner case.

Reply:
Good idea. I've done it and added unit testing of these `blockTo*` functions.